tracing/tracing.go

```diff
@@ -11,17 +11,31 @@ import (
 	"go.opentelemetry.io/otel/trace"
 )
 
-// InitTracer creates an OTel TracerProvider that exports the traces to a Jaeger collector.
-func InitTracer(serviceName, collectorEndpoint string, samplingFraction float64) (tp trace.TracerProvider, closer func(), err error) {
-	disabled := collectorEndpoint == ""
+// EndpointType represents the type of the tracing endpoint.
+const (
```

Comment:
I slightly nicer way to use Go's type system here is to declare a new string type and to make these consts specific instances of the new type, e.g.:
```golang
type EndpointType string
const (
    CollectorEndpointType EndpointType = "collector"
    AgentEndpointType EndpointType = "agent"
)